client/app/containers/index.js

```diff
@@ -1,5 +1,57 @@
 import React from 'react';
+import Main from '../components/';
 
-const App = () => <h1>Working</h1>;
+import * as api from '../api';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      directoryEntry: [],
+      path: '?path=.'
+    };
+
+    this.onDirectoryClick = this.onDirectoryClick.bind(this);
+  }
+
+  onDirectoryClick(link) {
+    const newPath = `${this.state.path}/${link}`;
+    
+    api.getTree(newPath).then(json => {
+      const newState = {
+        directoryEntry: json,
+        path: newPath
+      };
+
+      window.history.pushState(newState, null, newPath);
+      this.setState(newState);
+    });
+  }
+
+  componentDidMount() {
+    const newPath = window.location.search || this.state.path;
+
+    api.getTree(newPath).then(json => {
```

Comment:
Reccurent thing w/ fetch/forming new state and setting state can be extracted to method, imo